src/docs/introduction.rst

```diff
@@ -1,12 +1,102 @@
+instead, set up a trust and allow Bob from RedHat to administer
+the IBM domain…
+
+It’s not uncommon for companies to have two departments, one that supports
+and administers Linux and the other team does Windows, so each will have
+control over its own users and groups. Using SSSD’s ipa_provider,  AD domain
+users can then log in to a host managed by IPA because of the established
+trust between domains.
+
+SSSD can authenticate directly to AD, using the ad_provider, and they’ll
+be no need for IPA servers. Users and groups will be entirely managed by
+AD. So why use indirect authentication? IPA is more flexible for managing
+Linux hosts than AD since it can manage SUDO rules, SELinux users and
+contexts, automount maps, and more. If all users are created in AD, but
+the company has two IT teams. The Linux team may have to wait for users
+to be created. Hopefully, you can start to see that each solution has its
+pros and cons and will not be discussed in this introduction.
+
+For more information about FreeIPA and other compatible directory servers,
+please check out the following links.
+
+- `FreeIPA <https://www.freeipa.org/page/Main_Page>`_
+- `389DS <https://directory.fedoraproject.org/>`_
+- `OpenLDAP <https://www.openldap.org/>`_
+- `Kerberos <https://web.mit.edu/kerberos/>`_
+- `Microsoft ADDS <https://www.microsoft.com/en-ie/windows-server>`_
+
+The ad_provider and ipa_provider are just some providers that can be used
+for authentication or authorization. Other providers is the ldap_provider,
+krb_provider, proxy_provider, and local_provider. For more information please
+check out the following MAN pages for more information.
+
+- sssd-ldap
+- sssd-krb5
+- sssd-ipa
+- sssd-ad
+- sssd-files
+
+What else is important to know about SSSD? We’ve discussed authentication
+but we have not discussed authorization. Specific users and, or groups can
+be authorized to access the host using centrally managed HBACs
+*(Host-Based Access Control)* or filters in the SSSD configuration.
+For example, web admins don’t need access to all the servers on the network
+and only need access to the Apache and Nginx servers. This level of granular
+control can easily be done. The ad_provider can use AD’s GPOs
+*(Group Policy Objects)* for HBACs.
+
+In addition to access control, SSSD can lookup SUDO rules and roles to
+control what users can do and when they can do them on the system. For
+example, a web admin only needs access to stop, start, reboot the httpd
+service and edit the httpd configuration file. Any failed attempt to
+escalate their access will also be logged.
+
+Alright, this sounds great, but what happens when if the directory server
+is offline? Do I need to log in with a local user? Absolutely not, SSSD
+has a cache to improve performance after finding a user. The user’s
+password is also stored, as long as the user has logged in before, they
+will be able to login if SSSD when it’s unable to communicate to the servers.
```

Comment:
I would add one sentence about how this is useful for laptop systems configured to use SSSD, frequently reconnecting to wireless networks.

Reply:
Alright, this sounds great, but what happens when if the directory server 
is offline? **Or you have a poor wireless signal on your laptop and keep on 
disconnecting?** 

?

Reply:
Sure, that works.